assets/js/googlesitekit/widgets/index.js

```diff
@@ -19,10 +19,106 @@
 /**
  * Internal dependencies
  */
+import { dispatch, select } from 'googlesitekit-data';
+import { STORE_NAME } from './datastore/constants';
 // This import has side-effects; it registers the Widgets datastore on the default
 // data store registry (eg. `googlesitekit.data`).
 import './datastore';
 
-const Widgets = {};
+const Widgets = {
+	/**
+	 * Register a widget area
+	 *
+	 * @since n.e.x.t
+	 * @param {string}             slug                Widget Area's slug.
+	 * @param {Object}             settings            Widget Area's settings.
+	 * @param {string}             settings.title      Title for this widget area.
+	 * @param {string}             settings.subtitle   Subtitle for this widget area.
+	 * @param {number}             [settings.priority] Priority for this widget area. Default: 10.
+	 * @param {(string|undefined)} [settings.icon]     URL to SVG icon for this widget area.
+	 * @param {string}             [settings.style]    Widget area style (one of "boxes", "composite").
+	 * @param {(string|Array)}     [areaSlugs]         Optional. Widget area slug(s).
+	 */
+	registerWidgetArea( slug, settings, areaSlugs ) {
+		dispatch( STORE_NAME ).registerWidgetArea( slug, settings );
+		if ( areaSlugs ) {
+			dispatch( STORE_NAME ).assignWidget( slug, areaSlugs );
+		}
+	},
+
+	/**
+	 * Register a widget.
+	 *
+	 * @since n.e.x.t
+	 * @param {string}         slug           Widget's slug.
+	 * @param {Object}         settings       Widget's settings.
```

Comment:
The `settings` arguments should be documented here as well, similar like in the datastore and `registerWidgetArea` above:
```suggestion
	 * @param {string}          slug               Widget's slug.
	 * @param {Object}          settings           Widget's settings.
	 * @param {React.Component} settings.component React component used to display the contents of this widget.
	 * @param {number}          settings.priority  Optional. Widget's priority for ordering (lower number is higher priority, like WordPress hooks). Default is: 10.
	 * @param {string}          settings.width     Optional. Widget's maximum width to occupy. Default is: "quarter". One of: "quarter", "half", "full".
```